### PR TITLE
Add 'Managed by Puppet' template headers

### LINKED ISSUE #594: Add a "Managed by Puppet" header to configuration files
Hello everyone,

I was wondering why there is no "managed by puppet" header in configuration files. We use e.g.

```
#
# WARNING: This file is auto-generated by puppet!
# WARNING: To change it, edit the file in [wherever the file/template is]
#
```

I suggest adding such a header to all configuration files to clarify that Puppet is in fact managed by Puppet.

### PR DESCRIPTION
**Pull request by baurmatt**
Fixes #594.

### CODE REVIEW

**Comment by baurmatt** [comment]
Any update on this? This lays around for over 2 months now :)

File: templates/server/config.ru.erb [hunk 1]
```diff
@@ -1,5 +1,7 @@
-### Next part of the file is managed by a different template ###
-## Module:           '<%= scope.to_hash['module_name'] %>'
+#
+# WARNING: This file is auto-generated by puppet!
+# WARNING: To change it, edit the file in puppet/templates/server/config.ru.erb
```
Comment (mmoll) [inline templates/server/config.ru.erb]: I'm really unsure if it's a good idea to tell users to edit files... They should rather set parameters to chnage the generated files, IMHO.

**Comment by mmoll** [comment]
This also should get rebased, there are some more files in `templates/server/puppetserver/conf.d/` now.

File: templates/server/config.ru.erb (hunk 1 above)
Comment (baurmatt) [reply templates/server/config.ru.erb]: Good point, i've removed that line.

**Comment by baurmatt** [comment]
All done :)

**Comment by mmoll** [comment]
While personally I don't like these headers, this can go in, IMHO.

**Comment by ekohl** [comment]
I don't like these headers either. Especially the `Warning` part. In the context of our installer I'm not sure it really is a warning.

**Comment by hdep** [comment]
IMHO Those header are very usefull and almost a standard for puppet. In one shot you see that the file has been modified by puppet or not.

'Warning' or not isn't a big deal. Could be just  : 

'File managed by puppet'

**Comment by baurmatt** [comment]
I'm happy to change the wording, just tell we what it's supposed to say! :)

**Comment by ekohl** [comment]
I'd just stick with `Managed by puppet` as we already have in some places.

**Comment by baurmatt** [comment]
Done :)

File: spec/classes/puppet_server_config_spec.rb [hunk 2]
```diff
@@ -201,7 +201,6 @@
 
         it 'should contain autosign.conf with out content set' do
           should contain_file("#{confdir}/autosign.conf")
-          should_not contain_file("#{confdir}/autosign.conf").with_content(/# Managed by Puppet/)
```
Comment (ekohl) [inline spec/classes/puppet_server_config_spec.rb]: Could you keep this in but modified to the current header?

Comment (baurmatt) [reply spec/classes/puppet_server_config_spec.rb]: Done

**Comment by baurmatt** [comment]
* Rebased from master
* Removed exclamation mark
* Couldn't find any templates which get joined together by the following search criteria:
  * erb templates using template() function
  * concat::fragment's using template() function
  * template() function output gets concated with another template() function output

**Comment by mmoll** [comment]
@ekohl do we want to get this in?

**Comment by baurmatt** [comment]
(Hope this ends up on Github)

Will fix the merge conflict on monday morning :)

Send from one of my mobile devices

Am 11.01.2019 um 19:57 schrieb Ewoud Kohl van Wijngaarden <notifications@github.com <mailto:notifications@github.com> >:

@ekohl approved this pull request.

I guess this is OK, but there is now a merge conflict. One in the tests due to newlines, the other because we removed config.ru <http://config.ru> . Given I'm about to release I might fix it up myself on Monday if it hasn't by then.

**Comment by ekohl** [comment]
I fixed it up and merged it in f4e008e1546c2ecf091cb379766fc9e3f0148350. Thanks!

**Pull request closed without merging**